src/main/java/org/isf/medicalstock/gui/MovStockBrowser.java

```diff
@@ -932,6 +936,40 @@ private JButton getDischargeButton() {
 		return dischargeButton;
 	}
 
+	/**
+	 * This method creates the button that delete the last stock {@link Movement)
+	 * 
+	 * @return
+	 */
+	private JButton getDeleteLastMovementButton() {
+		JButton deleteMovementButton = new JButton(MessageBundle.getMessage("angal.common.delete.btn"));
+		deleteMovementButton.setMnemonic(MessageBundle.getMnemonic("angal.common.delete.btn.key"));
+		deleteMovementButton.setMnemonic(KeyEvent.VK_D);
+		deleteMovementButton.addActionListener(actionEvent -> {
+
+			int n = MessageDialog.yesNo(null, "angal.medicalstock.deletemovement.msg");
+
+			if (n == 0) {
+				try {
+					Movement lastMovement = movBrowserManager.getLastMovement();
+					if (lastMovement == null) {
+						MessageDialog.info(null, "angal.medicalstock.lastmovementnotfound.msg");
+						return;
+					} else {
```

Comment:
Here we have to check if the last movement is the selected movement and, if not, to block the deletion.
If you just got the lastMovement it can be null only if the database is empty, but in such case you would have nothing to delete...

Reply:
Sorry, there were too many things to suggest that I preferred to do it myself directly 😄
Please check the logic (that should be OK) and let's fix the CORE side.

Reply:
OK